src/test/java/org/apache/sling/resourceresolver/impl/mapping/ResourceMapperImplTest.java

```diff
@@ -110,6 +110,7 @@ public void prepare() throws LoginException {
         resourceProvider.putResource("/content/virtual/foo"); // matches virtual.host.com.80 mapping entry
         resourceProvider.putResource("/parent", PROP_ALIAS, "alias-parent"); // parent has alias
         resourceProvider.putResource("/parent/child", PROP_ALIAS, "alias-child"); // child has alias
+        resourceProvider.putResource("/parent/child/grandChild", PROP_ALIAS, "alias-grandChild"); // grandChild child has alias
```

Comment:
Does this reproduce only with 3+ path segments? I think we already have a scenario. with two segments

Reply:
Hi Robert,

**_>> Does this reproduce only with 3+ path segments?_**
No, this can also be reproduced with 2 segments, just to replicate the scenario mentioned in the SLING-10483, I added 3 segements.
But, I believe I should have used only 2 segments. WDYT?

**_>> I think we already have a scenario. with two segments_**
Yes, we have an already existing test with two segments (mapResourceWithNestedAlias), but this bug is reproducible only when we have all aliases segments present in the requested resource path.
And in the already existing test, the requested path does not consists of all aliases segments.